src/udx.c

```diff
@@ -1642,23 +1642,27 @@ udx_stream_destroy (udx_stream_t *handle) {
   // that creates some reentry trickiness incase this was called from on_read.
   clear_outgoing_packets(handle);
 
-  if (handle->relay_to == NULL) {
-    udx_packet_t *pkt = malloc(sizeof(udx_packet_t));
+  if (handle->relay_to != NULL) {
+    handle->status |= UDX_STREAM_DESTROYED;
+    close_maybe(handle, 0);
+    return 0;
+  }
 
-    uv_buf_t buf = uv_buf_init(NULL, 0);
+  udx_packet_t *pkt = malloc(sizeof(udx_packet_t));
```

Comment:
who frees this?

Reply:
The normal packet sending flow; the line was merely moved out of a conditional that is now handled above.

Reply:
ahhh, freaking github